frontend/ui-pablo/app/components/Organisms/swap/SwapForm.tsx

```diff
@@ -28,24 +28,49 @@ import { SwapRoute } from "./SwapRoute";
 import { PreviewModal } from "./PreviewModal";
 import { ConfirmingModal } from "./ConfirmingModal";
 import { useDotSamaContext, useParachainApi } from "substrate-react";
-import { Assets, AssetsValidForNow } from "@/defi/polkadot/Assets";
+import { Assets, AssetsValidForNow, getAssetOnChainId } from "@/defi/polkadot/Assets";
 import useStore from "@/store/useStore";
 import { AssetId } from "@/defi/polkadot/types";
 import { debounce } from "lodash";
-import { onSwapAmountChange } from "@/updaters/swaps/utils";
+import { calculateSwap } from "@/defi/utils/pablo/swaps";
+import { DEFAULT_NETWORK_ID } from "@/defi/utils";
 
 const SwapForm: React.FC<BoxProps> = ({ ...boxProps }) => {
   const isMobile = useMobile();
   const theme = useTheme();
   const dispatch = useDispatch();
   
-  const { parachainApi } = useParachainApi("picasso");
+  const { parachainApi } = useParachainApi(DEFAULT_NETWORK_ID);
   const { extensionStatus } = useDotSamaContext();
   
-  const { swaps, setUiAssetSelectionSwaps } = useStore();
+  const { swaps, setUiAssetSelectionSwaps, apollo, invertAssetSelectionSwaps } = useStore();
   const [valid1, setValid1] = useState<boolean>(false);
   const [valid2, setValid2] = useState<boolean>(false);
 
+  const token1PriceInUSD = useMemo(() => {
+    if (swaps.ui.baseAssetSelected === "none") return new BigNumber(0);
+    let base = getAssetOnChainId(DEFAULT_NETWORK_ID, swaps.ui.baseAssetSelected);
+    if (base) {
+      let baseId = base.toString()
+      if (apollo[baseId]) {
+        return new BigNumber(apollo[baseId])
+      }
+    }
+    return new BigNumber(0);
+  }, [swaps.ui.baseAssetSelected, apollo])
+
+  const token2PriceInUSD = useMemo(() => {
+    if (swaps.ui.quoteAssetSelected === "none") return new BigNumber(0);
+    let quote = getAssetOnChainId(DEFAULT_NETWORK_ID, swaps.ui.quoteAssetSelected);
+    if (quote) {
+      let quoteId = quote.toString()
+      if (apollo[quoteId]) {
+        return new BigNumber(apollo[quoteId])
+      }
+    }
+    return new BigNumber(0);
+  }, [swaps.ui.quoteAssetSelected, apollo])
+
```

Comment:
Suggestion for later, or if we have time, maybe now: 

```tsx
function getTokenPriceInUsd(token) { 
  if (token === "none") return new BigNumber(0);
    let quote = getAssetOnChainId(DEFAULT_NETWORK_ID, token)?.toString();
    return apollo[quoteId] ? return new BigNumber(apollo[quoteId] : return new BigNumber(0);
}

Reply:
can do, already have a hook.